### PR TITLE
Certs not removed with pki postgis on win port3

### PR DESCRIPTION
**Pull request by luipir**
## Description
port to 3.0 of #5477 with added unit tests.

ISSUE to resolve: On Windows, PKI PostGIS connections does not remove dumped and temporary certs on temp directory

For PKI db connections (at least Postgis) certs have to be dumped on files to setup connection and then removed.
File permissions are set here: https://qgis.org/api/2.18/qgsauthcertutils_8cpp_source.html#l00271
and python os.remove or QFile::remove(file) are enough on Linux and OSX. But not enough on Win because of ownership difference [1].
So it's necessary to explicitly setup minimum write permission to remove the file. The minimum is
QFile::WriteOwner.
In this moment only PostGIS provider is PKI enabled. The others DB, as far as I can read, still use basic auth.

This PR affect Postgis provider for c++ side and db_manager for the python side.

The issue need to be backported to 3.0.

[1] http://doc.qt.io/qt-4.8/qfile.html#Permission-enum

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [ ] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by luipir** [comment]
as in the closed PR, checking again how to reactivate postgres tests #5550

File: python/plugins/db_manager/db_plugins/postgis/connector.py [hunk 1]
```diff
@@ -66,7 +66,8 @@ def __init__(self, uri):
         try:
             self.connection = psycopg2.connect(expandedConnInfo)
         except self.connection_error_types() as e:
-            err = str(e)
+            # get credentials if cached or assking to the user no more than 3 times
+            err = unicode(e)
```
Comment (m-kuhn) [inline python/plugins/db_manager/db_plugins/postgis/connector.py]: `str()` is preferable for python3 (and I think this targets master?)

Comment (luipir) [reply python/plugins/db_manager/db_plugins/postgis/connector.py]: right!

**Comment by m-kuhn** [comment]
@luipir any new on this pull request? Is this targetted at 3.2 or 3.4?

**Comment by nyalldawson** [comment]
My 2c on this one - I think we should disable the failing test (pending the required travis/docker changes), and then merge the rest. It fixes a valid bug, and I'd rather not block the bug fix solely on the missing infrastructure to get the test to run on CI.

**Comment by luipir** [comment]
@nyalldawson I agree, test need a rework waiting a PKI enabled postgis. The PKI enabled postgis is not a great work, just do a setup programmatically instead of my old and fragile solution to push a prepared dbdata and pgconf in a the kartoza pg docker.
The question is if it would be a "responsability" of big companies (or institutions) interested in a reliable test infrastructure to finance someone to setup this docker.
I'm struggled about the above point... but the struggle is resolved by the lack of time because I'm doing more point cloud analysis than qgis code.

**Comment by luipir** [comment]
blacklisted the new test

**Comment by luipir** [comment]
no idea why spellcheck fail... prepare_commit.sh here does not generate any change!

**Comment by m-kuhn** [comment]
prepare_commit.sh doesn't do any spell check (unless you export `QGIS_CHECK_SPELLING=1`) by default
However the message on travis looks straightforward to fix as far as I can tell :)
https://travis-ci.org/qgis/QGIS/jobs/386084946#L740

**Comment by luipir** [comment]
travis is happy (but with blacklisted unit test for the fix)

**Review by m-kuhn — APPROVED** [review]

**Comment by m-kuhn** [comment]
Thanks at @luipir 
would be great if we could remind those organizations who requested and work with this feature of their duties.

**Comment by luipir** [comment]
one is Boundless, but I'm not the best person to contact them, and the second is at lease GeoSolutions because @giohappy faced this problem and now he's working there.
I suppose any other public institution (Tuscany or Piemonte region, Swiss administrations, ... ) should be interested to have a reliable test suite on pg.
IMHO this would be in charge of PSC.

**Comment by nyalldawson** [comment]
This can be merged now, right?

**Comment by giohappy** [comment]
Great. Nice to see this bug fixed! @luipir can you describe somewhere what is needed for a more realiable PKI enabled PostGIS docker?

**Comment by luipir** [comment]
@giohappy in an old comment of a closed issue:
https://github.com/qgis/QGIS/pull/6664#issuecomment-375878406

In that solution pgdata was prepared as binary instead to create it programmatically (great error sorry). The pgconf should be the same, but the users granted to access have to be created programmatically in db.

**Pull request merged**